### DISCUSSION TITLE
Link is already in database question

### DISCUSSION THREAD

**Discussion by jimojeda** [YOURLS/YOURLS#3168] [Q&A]
First of all, thank you for this amazing piece of software, I absolutely LOVE YOURLS!

Just one question that I haven't seen answered.

Whenever someone tries to create a short link and the target URL is already in the database, it warns that the target link is already in the database.

I know I can update the config file to turn off unique links.

But would it be possible instead to have YOURL say something like...

There already is a short link for that destination URL which is: [short link here]

That way, I don't have to worry that there are too many repetitive links in the database because I don't need too many links going to the same destination URL, know what I mean?

**Comment by dgw** [comment]
> know what I mean?

Not really. Either you want to allow multiple shortlinks to the same target, or not, and in either case you can search in the admin interface for all or part of the long URL to find what shortlinks exist for it.

What am I missing? Is this a feature request for a "YOURLS_UNIQUE_URLS_UNLESS_I_SAY_OTHERWISE" setting that will show you the existing keyword the first time you try to re-shorten a link but create a new one if you try again/specify a keyword/something else?

**Reply by jimojeda** [reply]
Correct... I don't want to allow multiple shortlinks to the same target URL.

So when you try to create a shortlink, if there's already a shortlink to that target URL, it should come back with a message like, "There's already a shortlink to that destination URL in the database.  Here's that link: https://example.com/short"

That way, I can use the shortened link that already exists instead of creating new ones each time that go to a destination URL that is already in the database.

I know that my installation is set like that already.  It's "unique links" feature is activated.

All I'm saying is that I'd like for the code to reveal the shortlink to the user trying to shorten a link to a target URL that already exists.

Does that make sense? (and please forgive me if I'm not wording things right.)

**Reply by dgw** [reply]
Ah, this is purely a front-end enhancement, though could be reflected in how the back-end returns the error if other maintainers like the idea.

Shouldn't be a big deal for anyone who knows basic JavaScript to patch it in here:

https://github.com/YOURLS/YOURLS/blob/42fd3b85c2d49120f3328c7f77ce4d5c100fcc61/js/insert.js#L27-L58

since the AJAX response should already contain the required info:

https://github.com/YOURLS/YOURLS/blob/42fd3b85c2d49120f3328c7f77ce4d5c100fcc61/includes/functions-shorturls.php#L129-L140
